### PR TITLE
Support for Doctrine Custom Mapping Types and Entity constructor with arguments

### PR DESCRIPTION
**Pull request by rvadym**
- Support for Doctrine Custom Mapping Types #38 
- Entity constructor with arguments

### CODE REVIEW

**Comment by Webonaute** [comment]
please, dont create pull request for 2 different feature in one.

For the constructor argument, can you provide a test case?

For the Custom mapping, I will look into it to see if there is no other way to avoid managing this by a try/catch exception.

thank.

**Comment by rvadym** [comment]
Hi
Yeah, you right, my solution for CMT is a bit ugly :) Just first what came to my mind. 

About constructor argument. I tried to describe it in proposed doc update in PR, which is [here](https://github.com/rvadym/DoctrineFixturesGeneratorBundle/blob/848f4cae1e13de776530dc2aea3879d4ff9e5126/Resources/doc/index.md#contructor-arguments)
Ask me if you still didn't get purpose of this update from docs, I'll try to be more specific. 

But I just faced another problem which is related to this issue - generated fixture class must have constructor arguments as well. 

The generated fixture file looks like this
```php
$item1 = new MonthlyProgress();
```
but in fact it must be like this
```
$item1 = new MonthlyProgress(\Money\Money::USD(0));
```
Trying to solve this right now.

File: Resources/doc/index.md [hunk 1]
```diff
@@ -64,6 +64,47 @@ It will create one file per entity you have in your project, it will create it i
...
+// src/AppBundle/Entity/Comment.php
+namespace AppBundle\Entity
+
+use AppBundle\Entity\Article;
+use Webonaute\DoctrineFixturesGeneratorBundle\Annotation\ConstructorArguments;
+
+/**
+ * @ConstructorArguments(
+ *     value={
+ *       "article" : {"php"  : "new \AppBundle\Entity\Article()"},
```
Comment (Webonaute) [inline Resources/doc/index.md]: I am not fan of this. It should be more transparent. It's too much to configure on each class who have constructor.

File: Tool/FixtureGenerator.php [hunk 2]
```diff
@@ -362,13 +401,17 @@ public function generateFixtureItemStub($item)
...
-                        //check reference.
-                        $relatedClass = get_class($value);
-                        $relatedEntity = ClassUtils::getRealClass($relatedClass);
-                        $identifiersIdsString = $this->getRelatedIdsForReference($relatedEntity, $value);
-                        $setValue = "\$this->getReference('{$this->referencePrefix}{$this->getEntityNameForRef($relatedEntity)}$identifiersIdsString')";
-                        $comment = "";
-
+                        try {
+                            //check reference.
+                            $relatedClass = get_class($value);
```
Comment (Webonaute) [inline Tool/FixtureGenerator.php]: For now I am good with this.

File: Tool/FixtureGenerator.php [hunk 3]
```diff
@@ -326,8 +331,42 @@ public function generateFixtureItemStub($item)
 
         $code = "";
         $reflexion = new \ReflectionClass($item);
+
+        /** @var array $constructorArguments */
+        $constructorArguments = [];
+
+        /** @var \ReflectionMethod $constructor */
+        if ($constructor = $reflexion->getConstructor()) {
```
Comment (Webonaute) [inline Tool/FixtureGenerator.php]: This should probably go inside it own getConstructorArguments() method.

File: Tool/FixtureGenerator.php [hunk 4]
```diff
@@ -326,8 +331,42 @@ public function generateFixtureItemStub($item)
...
+
+                    $constructorArguments[$k] = $caValue;
+                }
+            }
+        }
+
         $properties = $this->getRecursiveProperties($reflexion);
-        $newInstance = $reflexion->newInstance();
+
+        if (count($constructorArguments) > 0) {
```
Comment (Webonaute) [inline Tool/FixtureGenerator.php]: this need to go in its own method. getNewInstanceFromReflexion($reflexion, $constructorArguments).

**Comment by Webonaute** [comment]
sorry for the delay. I am very busy these day.

**Comment by Webonaute** [comment]
I just accept a pull request for constructor handling.

**Pull request closed without merging**